core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlowConfig.scala

```diff
@@ -21,15 +23,45 @@ import scala.concurrent.duration._
   * @param commitOffsetsInterval
   *   How often the state is to be evaluated for the pending commits.
   *
-  * @param parallelRecovery
-  *   If `true` tries to recover the keys in parallel. Note that it, currently, requires for all keys to fit in memory.
+  * @param recoveryMode
+  *   Controls how the snapshots are recovered. The default mode is `ParallelUnbounded`, which is the fastest, but
+  *   requires all the keys to fit in memory and might lead to CPU starvation if there are a lot of keys. See
+  *   [[com.evolutiongaming.kafka.flow.PartitionFlowConfig.RecoveryMode$]] for more details.
   *
   * @param commitOnRevoke
-  *   Try commiting everything when partition is revoked.
+  *   Try committing everything when partition is revoked.
   */
 case class PartitionFlowConfig(
   triggerTimersInterval: FiniteDuration = 1.second,
   commitOffsetsInterval: FiniteDuration = 1.minute,
-  parallelRecovery: Boolean             = true,
+  recoveryMode: RecoveryMode            = RecoveryMode.ParallelUnbounded,
   commitOnRevoke: Boolean               = false
 )
+
+object PartitionFlowConfig {
+
+  sealed trait RecoveryMode
+
+  object RecoveryMode {
+
+    /** Read snapshots in parallel with a limit on the number of fibers spawned. For each key a fiber will be spawned,
+      * but a total number of concurrent fibers will not exceed the specified limit. This mode is slower than
+      * `ParallelUnbounded`, but it allows fine-tuning the number of concurrent fibers to prevent CPU starvation. Just
```

Comment:
> but it allows fine-tuning the number of concurrent fibers to prevent CPU starvation

 but it allows fine-tuning the number of concurrent fibers to prevent CPU starvation and overwhelming the underlying storage with too many parallel recoveries.